server.js

```diff
@@ -9,34 +9,25 @@ var config = require('./config');
 var server = express();
 
 var api = require('./index');
+var middlewareAuthentication = require('./middleware/authentication');
+var middlewareCORS = require('./middleware/cors');
 
 mongoose.connect(config.mongodb_uri, (err) => {
 
-    // CORS
-    server.use((_, res, next) => {
-        // Website you wish to allow to connect
-        res.setHeader('Access-Control-Allow-Origin', '*');
-        // Request methods you wish to allow
-        res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS, PUT, PATCH, DELETE');
-        // Request headers you wish to allow
-        res.setHeader('Access-Control-Allow-Headers', 'X-Requested-With,content-type');
-        // Set to true if you need the website to include cookies in the requests sent
-        // to the API (e.g. in case you use sessions)
-        res.setHeader('Access-Control-Allow-Credentials', true);
-        // Pass to next layer of middleware
-        next();
-    });
+    if(typeof config.auth_token === 'undefined') {
+        throw 'Missing environment token';
+    }
 
+    server.use(middlewareCORS);
     server.use(morgan(':method :url | :status | :response-time ms'));
     server.use(express.static(__dirname));
-    server.use(bodyparser.json({limit: '50mb'}));
+    server.use(bodyparser.json({
+        limit: '50mb'
+    }));
+    server.use('/api/', middlewareAuthentication);
```

Comment:
Why are you protecting all the endpoints? We just want to protect the sensitive endpoints (admin ones)